pallets/rewards/src/migrations.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use frame_support::traits::{
+	fungibles::{Inspect, Mutate},
+	Get, OnRuntimeUpgrade,
+};
+use sp_runtime::traits::AccountIdConversion;
+use sp_std::marker::PhantomData;
+#[cfg(feature = "try-runtime")]
+use {frame_support::inherent::Vec, sp_std::vec};
+
+use crate::{BalanceOf, Config};
+
+pub struct FundExistentialDeposit<T, I, Currency, ED>(PhantomData<(T, I, Currency, ED)>);
+
+fn pallet_is_funded<T, I, Currency, ED>() -> bool
+where
+	T: Config<I>,
+	I: 'static,
+	Currency: Get<<T as Config<I>>::CurrencyId>,
+	ED: Get<BalanceOf<T, I>>,
+{
+	let balance = <<T as Config<I>>::Currency as Inspect<T::AccountId>>::balance(
+		Currency::get(),
+		&<T as Config<I>>::PalletId::get().into_account_truncating(),
+	);
+	balance >= ED::get()
+}
+
```

Comment:
Could it be possible that an account already has funding from rewards and we still would need to fund 1 times ED in case all collators claim?

Reply:
Funding the pallet's sovereign account once seems to be sufficient based on the unit tests and my local e2e testing. When we update the rewards, they are associated to a group. This ensures that a group cannot consume more than it should. 

However, before the migration, the balance of the pallet's sovereign account only consisted of the block rewards. So it was impossible to claim all rewards.